src/rules/variable-validate-interpolation/index.js

```diff
@@ -0,0 +1,127 @@
+import { utils } from "stylelint"
+import valueParser from "postcss-value-parser"
+
+export const ruleName = "variable-validate-interpolation"
+
+export const messages = utils.ruleMessages(ruleName, {
+  rejected: (n, v) => `You need to interpolate your variable ${v} when using it with ${n}`,
```

Comment:
FYI, messages in stylelint [start with](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#determine-warning-messages) "Expected ... " or "Unexpected ..." e.g. `"Expected variable \"${variable}\" to be interpolated when using it with \"${node}\""`